### PR TITLE
Implement more OpenSSL compatibility functions

### PR DESCRIPTION
**Pull request by julek-wolfssl**
- i2d_PKCS8PrivateKey_bio
- X509V3_EXT_i2d
- SSL_renegotiate_pending

### CODE REVIEW

**Comment by kaleb-himes** [comment]
retest this please.

File: src/ssl.c [hunk 1]
```diff
@@ -25454,6 +25719,35 @@ WOLFSSL_API int wolfSSL_i2a_ASN1_OBJECT(WOLFSSL_BIO *bp,
...
+    else {
+        p = (byte*)XMALLOC(a->objSz, NULL, DYNAMIC_TYPE_OPENSSL);
+        if (!p) {
+            WOLFSSL_MSG("Bad malloc");
+            return WOLFSSL_FAILURE;
+        }
+    }
+
+    XMEMCPY(p, a->obj, a->objSz);
+    *pp = p + a->objSz;
```
Comment (JacobBarthelmeh) [inline src/ssl.c:25747]: could dereference null pointer here

File: src/ssl.c [hunk 2]
```diff
@@ -25454,6 +25719,35 @@ WOLFSSL_API int wolfSSL_i2a_ASN1_OBJECT(WOLFSSL_BIO *bp,
     return WOLFSSL_FAILURE;
 }
 
+int wolfSSL_i2d_ASN1_OBJECT(WOLFSSL_ASN1_OBJECT *a, unsigned char **pp)
```
Comment (JacobBarthelmeh) [inline src/ssl.c:25722]: please add a comment, at the minimum having return value expected

File: src/ssl.c (hunk 1 above)
Comment (JacobBarthelmeh) [reply src/ssl.c:25747]: disregard. This is checked on line 25733 and a->objSz is returned before this section of code is hit

**Review by JacobBarthelmeh — APPROVED** [review]

File: src/ssl.c (hunk 2 above)
Comment (JacobBarthelmeh) [reply src/ssl.c:25722]: Not critical enough to restart Jenkins test for this alone.

**Review by toddouska — APPROVED** [review]

**Pull request merged**